### ISSUE TITLE
Text Search does not work when string can be transformed to date

### ISSUE THREAD

**Issue by psyfreak** [jupe/mongoose-query#146]
A query with a text search does not work if the phrase results in a date ( if this !Number.isNaN(date) is true).

Try:  
q (exact phrase | and)  { '$text': { '$search': '"100-10-7"' } }  or 
q ( or )  { '$text': { '$search': '100-10-7' } } 

In my example I wanted to search within a textIndex for a cas number which results to be transformed into this iso date: 0100-10-06T23:06:32.000Z

Orginal code in parseQuery.js in function jsonConverter 

```
        // check if date to convert it
        const date = parseDateCustom(value);
        if (!Number.isNaN(date)) {
          // eslint-disable-next-line no-param-reassign
          obj[key] = date;
          // the phrase '"100-10-7"' is transformed to a date.
        } else {
          // eslint-disable-next-line no-param-reassign
          obj[key] = value;
        }
```

Possible Solution
add condition to 
```
if (key === '$regex') {
  ...
}
else if (/^oid:[a-fA-F0-9]{24}$/.test(value)) {
      ...
}
// new block
else if(key === '$search') {
   obj[key] = value;
}
// end new block
else {
...
}
```

I can submit a pr if you want. Thank you for you work.

**Comment by jupe** [comment]
Thank for the report. Are you able to contribute fix with tests?

**Comment by jupe** [comment]
oh, you did it already. Great🙏

**Comment by jupe** [comment]
fixed by #147
